### ISSUE TITLE
dns auto_search isn't working

### ISSUE THREAD

**Issue by hjerning** [qutebrowser/qutebrowser#7849]
**Version info**:
```
qutebrowser v3.0.0
Git commit: 
Backend: QtWebEngine 6.5.2, based on Chromium 108.0.5359.220 (from api)
Qt: 6.5.2

CPython: 3.11.3
PyQt: 6.5.2

Qt wrapper info:
  PyQt6: success
  PyQt5: not imported
  -> selected: PyQt6 (via autoselect)

colorama: 0.4.6
jinja2: 3.1.2
pygments: no
yaml: 6.0.1
adblock: 0.6.0
objc: no
PyQt6.QtWebEngineCore: 6.5.0
PyQt6.sip: 6.7.10
pdf.js: no
sqlite: 3.42.0
QtNetwork SSL: OpenSSL 3.1.2 1 Aug 2023

Style: QFusionStyle
Platform plugin: xcb
OpenGL: Intel, 4.6 (Compatibility Profile) Mesa 23.1.6
Platform: Linux-6.1.46-1-lts-x86_64-with-glibc2.38, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.11/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt6, data path: /usr/share/qt6
```

**Does the bug happen if you start with `--temp-basedir`?**: Yes

**Description**

With `url.auto_search` configured to `dns` version 3.0.0 will even treat a FQDN as something to send to the searchengine.
`naive` works as expected with a FQDN but local hostnames will no longer be resolved by either `naive` or `dns`.
 
**How to reproduce**
1. `qutebrowser --temp-basedir`
2. `:set url.auto_search dns`
3. `:open www.qutebrowser.org`

**Comment by The-Compiler** [comment]
Confirmed. With `qutebrowser --debug --temp-basedir --qt-wrapper PyQt5 -s url.auto_search dns www.qutebrowser.org`:

```
16:35:17 DEBUG    url        urlutils:is_url:310 Checking if 'www.qutebrowser.org' is a URL (autosearch=dns).
16:35:17 DEBUG    url        urlutils:is_url:348 Checking via DNS check
16:35:17 DEBUG    url        urlutils:_is_url_dns:227 Doing DNS request for www.qutebrowser.org
16:35:17 DEBUG    url        urlutils:is_url:357 url = True
16:35:17 DEBUG    url        urlutils:fuzzy_url:264 URL is a fuzzy address
16:35:17 DEBUG    url        urlutils:fuzzy_url:266 Converting fuzzy term 'www.qutebrowser.org' to URL -> http://www.qutebrowser.org
16:35:17 DEBUG    init       app:open_url:288 About to open URL: http://www.qutebrowser.org
```

but with `--qt-wrapper PyQt6`:

```
16:36:02 DEBUG    url        urlutils:is_url:310 Checking if 'www.qutebrowser.org' is a URL (autosearch=dns).
16:36:02 DEBUG    url        urlutils:is_url:348 Checking via DNS check
16:36:02 DEBUG    url        urlutils:_is_url_dns:227 Doing DNS request for www.qutebrowser.org
16:36:02 DEBUG    url        urlutils:is_url:357 url = False
16:36:02 DEBUG    url        urlutils:fuzzy_url:258 URL is a fuzzy search term
16:36:02 DEBUG    url        urlutils:_get_search_url:157 Finding search engine for 'www.qutebrowser.org'
16:36:02 DEBUG    url        urlutils:_parse_search_term:144 engine None, term 'www.qutebrowser.org'
16:36:02 DEBUG    url        urlutils:fuzzy_url:266 Converting fuzzy term 'www.qutebrowser.org' to URL -> https://duckduckgo.com/?q=www.qutebrowser.org
16:36:02 DEBUG    init       app:open_url:288 About to open URL: https://duckduckgo.com/?q=www.qutebrowser.org
```

**Comment by The-Compiler** [comment]
Fixed in 47daa9d.
